Robot2018/src/org/usfirst/frc/team199/Robot2018/subsystems/Lift.java

```diff
@@ -83,15 +129,65 @@ public void runMotor(double output) {
 	 * 
 	 * @return pos - current position
 	 */
-	public Position getCurrPos() {
+	@Override
+	public LiftHeight getCurrPos() {
 		return currPosition;
 	}
 
 	/**
 	 * Resets the encoder
 	 */
+	@Override
 	public void resetEnc() {
 		liftEnc.reset();
 	}
 
+	/**
+	 * Gets the number of stages variable
+	 */
+	@Override
+	public int getNumStages() {
+		return NUM_STAGES;
+	}
+
+	/**
+	 * Gets the extra distance above the switch or scale we want to lift in inches
+	 */
+	@Override
+	public double getWiggleRoom() {
+		return WIGGLE_ROOM;
+	}
+
+	/**
+	 * Uses AMT103 Encoder to detect the current lift height with respect to the
+	 * lift's min height (inches)
+	 */
+	@Override
+	protected double returnPIDInput() {
+		// return getHeight(); //Use this instead? What's the difference?
+		return liftEnc.getDistance();
+	}
+
+	/**
+	 * Runs the lift motors at the value that the pid loop calculated. Used
+	 * internally by PIDSubsystem.
+	 */
+	@Override
+	protected void usePIDOutput(double output) {
+		double out = output;
+		double spd = liftEnc.getRate();
+		out += convertSpdToVoltage(spd);
```

Comment:
I was wrong about this. You should be able to just add a constant configurable value. It shouldn't change with speed.